src/main/java/io/gravitee/repository/redis/common/RedisConnectionFactory.java

```diff
@@ -84,4 +111,21 @@ private String readPropertyValue(String propertyName) {
         logger.debug("Read property {}: {}", propertyName, value);
         return value;
     }
+
+    private LettucePoolingClientConfiguration buildLettuceClientConfiguration() {
+        final LettucePoolingClientConfigurationBuilder builder = LettucePoolingClientConfiguration.builder();
+        builder.commandTimeout(Duration.ofMillis(readPropertyValue(propertyPrefix + "timeout", int.class, 1000)));
```

Comment:
Previous version of the timeout was  `-1`.
What can be the impact for current user which are using this `-1` as the default value as part of their `gravitee.yml` ?

Reply:
Lettuce client does not support negative value for Timeout : 
https://github.com/lettuce-io/lettuce-core/blob/b978238db8a33c8860819b73e528f016f940e992/src/main/java/io/lettuce/core/RedisChannelHandler.java#L103

Setting the timeout to `0` (for no timeout) does not work either : this is literally interpreted as a 0 second timeout to execute a command. Perhaps a bug ? I will try to investigate a little more on this point, if I have time ...

Personnaly, I'm not a big fan of a default timeout to 'no timeout', especially regarding a Redis command.

A user who would have explicitly set this parameter to `-1` in his gravitee.yml would get an error during startup of the gateway.
On the other hand, if he used the default timeout (no value for the timeout parameter in gravitee.yml), it will be transparent.

We can add a safety check to convert the negative timeout to the default one to ensure compatibility.

What do you think ?

Reply:
LGTM 👍